api/BUILD

```diff
@@ -208,6 +208,7 @@ proto_library(
         "//envoy/extensions/filters/http/ratelimit/v3:pkg",
         "//envoy/extensions/filters/http/rbac/v3:pkg",
         "//envoy/extensions/filters/http/router/v3:pkg",
+        "//envoy/extensions/filters/http/simple_http_cache/v3alpha:pkg",
```

Comment:
I think this lives under `//source/extensions/filters/http/cache/v3/simpler_http_cache/v3alpha`. @jmarantz @toddmgreer thoughts?

Reply:
Envoy's source layout policies are too mysterious for me to say anything with confidence, but in hopes that I'll at least be wrong in a clarifying way...

SimpleHttpCache is the first plug-in extension of CacheFilter--envoy.source.extensions.filters.http.cache.SimpleHttpCacheConfig exists in order to be put into envoy.extensions.filters.http.cache.v3alpha.CacheConfig.typed_config.

Since CacheFilter's config message is in //api/envoy/extensions/filters/http/cache/v3alpha/cache.proto, maybe SimpleHttpCache's config message should be in //api/envoy/extensions/filters/http/cache/v3alpha/plugins/simple_http_cache.proto?

Reply:
thanks thats very helpful - the relationship between these two had so far eluded me

based on this im realising that:

- `envoy.filters.http.cache` (or similar) should be an "extension category" 
- `simple_http_cache` should have its extension category set to ^^ (its currently set to `envoy.filters.http`)

ill experiment with fixing this as a first step i think

the thing that this also highlights is the lack of reference documentation for these filters/extensions - i have opened a ticket here #16246. Im not sure that im sufficiently knowledgeable about its workings atm - so any help filling in the blanks here would be greatly appreciated

Reply:
PR added here #16249 to try and address the cache cat confusions

Reply:
after messing around with this in #16249 ive got something that kinda works - ive closed that PR and moved changes here